### PR TITLE
Fixed minor issues with upgrade guide

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

Fixed:
- I removed anchors from headers, they are autogenerated. No need for manual ones. It also prevents the generation of the right-side menu
- One occasion of migrating fixed to upgrade
- Fixed a link to support to actually point at support

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

### CODE REVIEW

**Comment by artemkovalyov** [comment]
image

File: docs/js/guides/upgrade-to-version-2.mdx [hunk 1]
```diff
@@ -13,31 +13,31 @@ This document will guide you through the steps necessary to upgrade to version 2
...
-- [Update Generated API Clients](#updateGeneratedClient)
-- [Replace Core Imports](#replaceCoreImports)
-- [Adjust Serializing and Deserializing of Entities](#adjustSerializer)
-- [Get Support if you face a problem in the migration process](#getSupport)
+- [Upgrade ECMAScript runtime to "es2019" or higher](#update-ecmascript-runtime)
+- [Update Public API Clients - possible after beta phase](#update-public-api-clients)
+- [Update Generated API Clients](#update-generated-api-clients)
+- [Replace Core Imports](#replace-core-imports)
+- [Adjust Serializing and Deserializing of Entities](#adjust-serializing-and-deserializing-of-entities)
+- [Get Support if you face a problem in the upgrade process](get-support)
```
Comment (deekshas8) [inline docs/js/guides/upgrade-to-version-2.mdx:21]:
```suggestion
- [Get Support if you face a problem in the upgrade process](#get-support)
```

File: docs/js/guides/upgrade-to-version-2.mdx [hunk 2]
```diff
@@ -71,7 +71,7 @@ We assume that you have installed the generator as a local `devDependency` in yo
 - Change the `requestBuillder()` call as described [here](#newRequestApi)
```
Comment (deekshas8) [inline docs/js/guides/upgrade-to-version-2.mdx:71]:
```suggestion
- Change the `requestBuillder()` call as described [here](#breaking-change-on-the-requestbuilder-api)
```

File: docs/js/guides/upgrade-to-version-2.mdx [hunk 3]
```diff
@@ -71,7 +71,7 @@ We assume that you have installed the generator as a local `devDependency` in yo
...
 - Change the `execute()` call as described [here](#executeAPIChanges)
```
Comment (deekshas8) [inline docs/js/guides/upgrade-to-version-2.mdx:72]:
```suggestion
- Change the `execute()` call as described [here](#breaking-change-on-the-execute-api)
```

**Comment by artemkovalyov** [comment]
Thanks for catching more links, @deekshas8. Take another look, please.

**Review by deekshas8 — APPROVED** [review]
Pushed a fix for few other links. 
LGTM.

**Pull request merged**